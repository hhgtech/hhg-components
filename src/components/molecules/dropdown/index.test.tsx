import React from 'react'

import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { Dropdown } from '.'
import MockTheme from './../../../mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const dropdown = render(
    <MockTheme>
      <Dropdown
        id="dropdown1"
        placeholder="Dropdown"
        placeHolderIcon={<img src="//unsplash.it/16" />}
        value={undefined}
        options={[
          {
            label: 'None',
            value: '1',
            icon: <img src="//unsplash.it/16" />,
          },
          {
            label: 'Second point',
            value: '2',
            icon: <img src="//unsplash.it/16" />,
          },
          {
            label: 'Third point',
            value: '3',
            icon: <img src="//unsplash.it/16" />,
          },
        ]}
      />
    </MockTheme>,
  )

  const self = dropdown.container.firstChild

  expect(self).toMatchSnapshot()

  expect(self).toContainHTML('<img')
})
