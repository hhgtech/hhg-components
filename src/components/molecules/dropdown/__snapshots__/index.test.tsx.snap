// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
  border: 1px solid #d9d9d9;
  background: url("data:image/svg+xml,%3Csvg width='10' height='6' viewBox='0 0 10 6' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M1 1L5 5L9 1' stroke='%23595959' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A") no-repeat;
  background-position: right 16px center;
  border-radius: 6px;
}

.c0[data-has-icons='true'] {
  border: none;
  background: #f2f2f2;
}

.c0[data-has-icons='true'] img,
.c0[data-has-icons='true'] svg {
  display: inline-block;
  max-width: 16px;
  margin-right: 6px;
}

.c0[data-has-icons='true'] span {
  vertical-align: middle;
  display: inline-block;
}

.c0[data-has-icons='true']:hover {
  box-shadow: none;
}

.c0[data-has-icons='true'] > div {
  position: relative;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
}

.c0[data-has-icons='true'] > div::after {
  content: '';
  right: 12px;
  top: 50%;
  background-image: url("data:image/svg+xml,%3Csvg width='10' height='6' viewBox='0 0 10 6' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M1 1L5 5L9 1' stroke='%23595959' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A");
  width: 10px;
  height: 8px;
  position: absolute;
  background-repeat: no-repeat;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0:hover {
  box-shadow: 0 0 2px 2px #91caff;
}

.c1 {
  padding: 12px 18px 12px 16px;
  cursor: pointer;
}

.c1[data-dropdown-open] {
  padding: 11px 15px 11px 15px;
  border: 1px solid #2d87f3;
  border-radius: 6px;
  box-shadow: 0 0 2px 2px #91caff;
}

<div
  class="c0"
  data-has-icons="true"
  id="dropdown1"
  tabindex="0"
>
  <div
    class="c1"
  >
    <img
      src="//unsplash.it/16"
    />
     
    <span>
      Dropdown
    </span>
  </div>
</div>
`;
